service/adrs/0006-optional-value-for-code-defaults.md

```diff
@@ -0,0 +1,102 @@
+2. Return an error response when they want to defer to code defaults (misleading telemetry and error handling)
+
+Neither approach is ideal. Returning errors for what is actually a successful evaluation leads to incorrect metrics and may trigger error handling logic in applications.
+
+## Decision
+
+Introduce a new flag type schema `codeDefaultFlag` that explicitly indicates the provider should use the default value defined in the application code. This new type would be used alongside existing flag types (booleanFlag, stringFlag, etc.) in evaluation responses.
+
+Key implementation details:
+- Add a new `codeDefaultFlag` schema type with no `value` field
+- Existing flag type schemas remain unchanged (backward compatible)
+- When a response uses `codeDefaultFlag` type, providers MUST use the code default value
+- All other fields in the evaluation response (key, reason, variant, metadata) remain unchanged
+- This behavior applies to both single flag evaluations and bulk evaluations
+- The `oneOf` discriminator in evaluation responses would include this new type
+
+Example OpenAPI schema addition:
+```yaml
+codeDefaultFlag:
+  description: A flag evaluation that defers to the code default value
+  type: object
+  # Note: No value property - provider must use code default
+```
+
+The evaluation success response would be updated to include this type:
+```yaml
+- oneOf:
+    - $ref: "#/components/schemas/booleanFlag"
+    - $ref: "#/components/schemas/stringFlag"
+    - $ref: "#/components/schemas/integerFlag"
+    - $ref: "#/components/schemas/floatFlag"
+    - $ref: "#/components/schemas/objectFlag"
+    - $ref: "#/components/schemas/codeDefaultFlag"  # New type
+```
+
+Example response deferring to code default:
+```json
+{
+  "key": "new-feature",
+  "reason": "DEFAULT",
+  "variant": "control"
+  // No value field - indicates codeDefaultFlag type
+}
+```
+
+Example response with explicit value (existing behavior):
+```json
+{
+  "key": "new-feature",
+  "value": true,
+  "reason": "TARGETING_MATCH",
+  "variant": "treatment"
+}
+```
+
+## Consequences
+
+### Positive
+
+- **Eliminates default value duplication**: Developers maintain default values in one place (code) rather than synchronizing between code and flag management system
+- **Supports standard feature flag lifecycle**: Natural progression from code defaults to managed values
+- **Accurate telemetry**: Deferred evaluations are correctly recorded as successful rather than errors
+- **Flexibility for flag management systems**: Systems can choose when to provide explicit values vs. deferring to code
+- **Backward compatible**: Existing implementations continue to work without modification
+
+### Negative
+
+- **Breaking change for providers**: While the API remains backward compatible, all provider implementations must be updated to handle the new `codeDefaultFlag` type. This change should be implemented before the 1.0 release.
```

Comment:
Implementations, I'm sure, already throw (and therefore cause the return of the default) if no value is included in the response. I think so the actual breakage here is fairly minor, and arguably not even truly a breakage, since at least in our situation at Dynatrace we are literally using errors to achieve the desired behavior at the moment.